plugins/jcache/src/main/java/org/radargun/service/JCacheLifecycle.java

```diff
@@ -0,0 +1,69 @@
+package org.radargun.service;
+
+import org.radargun.logging.Log;
+import org.radargun.logging.LogFactory;
+import org.radargun.traits.Lifecycle;
+import org.radargun.utils.Utils;
+
+import javax.cache.CacheManager;
+import javax.cache.Caching;
+import javax.cache.configuration.MutableConfiguration;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.Properties;
+
+/**
+ * @author Matej Cimbora
+ */
+public class JCacheLifecycle implements Lifecycle {
+
+   private static final Log log = LogFactory.getLog(JCacheLifecycle.class);
+   private JCacheService service;
+   private CacheManager cacheManager;
+
+   public JCacheLifecycle(JCacheService service) {
+      this.service = service;
+   }
+
+   @Override
+   public void start() {
+      URI configFileUri = null;
+      try {
+         if (service.configFile != null) {
+            URL configFile = JCacheLifecycle.class.getClassLoader().getResource(service.configFile);
+            if (configFile == null) {
+               throw new IllegalArgumentException(String.format("Unable to locate specified configuration file %s.", service.configFile));
+            }
+            configFileUri = configFile.toURI();
+         }
+      } catch (URISyntaxException e) {
+         log.errorf(String.format("Exception while accessing configuration file %s.", service.configFile), e);
+         throw new IllegalStateException(e);
+      }
+      Properties configProperties = null;
+      if (service.propertiesFile != null) {
+         configProperties = new Properties();
+         try {
+            Utils.loadProperties(configProperties, JCacheLifecycle.class.getClassLoader().getResourceAsStream(service.propertiesFile));
+         } catch (IOException e) {
+            log.error(String.format("Exception while loading configuration properties file %s.", service.propertiesFile));
+            throw new IllegalStateException(e);
+         }
+      }
+      cacheManager = Caching.getCachingProvider().getCacheManager(configFileUri, JCacheLifecycle.class.getClassLoader(), configProperties);
```

Comment:
Recent Infinispan versions bring JCacheProvider for both embedded and remote, and since there can (and should!) be both of them on classpath, you should add property for selection of the provider.